src/ATen/function_wrapper.py

```diff
@@ -75,6 +75,22 @@
     return static_cast<Type*>(this)->${method_prefix}${api_name}(${sparse_actuals});
 }""")
 
+WRAP_DIM_GEN = CodeTemplate("""\
+auto ${target}${dim}_ = std::max<${type}>(${target_dim_expr}, 0);
+if (${target}${dim}_ <= 0) {
```

Comment:
nit: `== 0` instead of `<= 0` ? the above code can never generate a negative number.

Reply:
I tend to slightly prefer the more defensive code (the code above can change and this won't break, at some very slight cost to readability).